pkg/specgen/generate/namespaces.go

```diff
@@ -29,20 +31,46 @@ func GetDefaultNamespaceMode(nsType string, cfg *config.Config, pod *libpod.Pod)
 	nsType = strings.ToLower(nsType)
 
 	// If the pod is not nil - check shared namespaces
+	//	pod.NetworkMode()
 	if pod != nil && pod.HasInfraContainer() {
 		podMode := false
 		switch {
+		case nsType == "pid" && pod.NamespaceMode(spec.PIDNamespace) == host:
```

Comment:
This is not correct, you have to do this inside the sharesPID case below, same for all the other namespace
see https://github.com/containers/podman/pull/14461#discussion_r888941070

Reply:
ohhhh I misread your comment, I thought it said more it before the host checking case. This make more sense, thanks

Reply:
Wait @Luap99 if I move these checks inside of `Shares...` for the example `podman pod create --net host` and then `podman run -dt --pod=<POD>` won't that give me a private network?

Reply:
yeah, that undoes the point of this PR. the original issue #13763 we decided we need to catch if we are in the host NS before we check for sharing because when you create a pod with `--net=host` pod.SharesNet is always false meaning this host condition will be missed

Reply:
no ShareXXX checks if the containers in the pod should share the same ns (`--share` option)

Reply:
remove this incorrect workaround: https://github.com/containers/podman/blob/864d46c77ef76241b6f0e0c725a08ec28e7e32bc/pkg/specgen/generate/namespaces.go#L495